cas-server-core-services/src/test/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDaoTests.java

```diff
@@ -0,0 +1,391 @@
+        r.setServiceId("^https://.+");
+        r.setName("testServiceType");
+        r.setTheme("testtheme");
+        r.setEvaluationOrder(1000);
+
+        final RegisteredService r2 = this.dao.save(r);
+        assertTrue(r2 instanceof  RegexRegisteredService);
+    }
+
+    @Test(expected=RuntimeException.class)
+    public void verifyServiceWithInvalidFileName() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("hell/o@world:*");
+        r.setEvaluationOrder(1000);
+
+        final RegisteredService r2 = this.dao.save(r);
+    }
+    
+    @Test
+    public void verifyServiceRemovals() throws Exception{
+        final List<RegisteredService> list = new ArrayList<>(5);
+        IntStream.range(1, 5).forEach(i -> {
+            final RegexRegisteredService r = new RegexRegisteredService();
+            r.setServiceId("^https://.+");
+            r.setName("testServiceType");
+            r.setTheme("testtheme");
+            r.setEvaluationOrder(1000);
+            r.setId(i * 100);
+            list.add(this.dao.save(r));
+        });
+
+        list.stream().forEach(r2 -> {
+            try {
+                Thread.sleep(500);
+                this.dao.delete(r2);
+                Thread.sleep(2000);
+            } catch (final InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            assertNull(this.dao.findServiceById(r2.getId()));
+        });
+    }
+
+    @Test
+    public void checkForAuthorizationStrategy() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("checkForAuthorizationStrategy");
+        r.setId(42);
+
+        final DefaultRegisteredServiceAccessStrategy authz =
+                new DefaultRegisteredServiceAccessStrategy(false, false);
+
+        final Map<String, Set<String>> attrs = new HashMap<>();
+        attrs.put("cn", Sets.newHashSet("v1, v2, v3"));
+        attrs.put("memberOf", Sets.newHashSet(Arrays.asList("v4, v5, v6")));
+        authz.setRequiredAttributes(attrs);
+        r.setAccessStrategy(authz);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+        assertEquals(r2, r3);
+    }
+
+    @Test
+    public void verifyAccessStrategyWithStarEndDate() throws Exception {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("verifyAAccessStrategyWithStarEndDate");
+        r.setId(62);
+
+        final TimeBasedRegisteredServiceAccessStrategy authz =
+                new TimeBasedRegisteredServiceAccessStrategy(true, false);
+
+        authz.setStartingDateTime(ZonedDateTime.now(ZoneOffset.UTC).plusDays(1).toString());
+        authz.setEndingDateTime(ZonedDateTime.now(ZoneOffset.UTC).plusDays(10).toString());
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['10' is a magic number.](https://www.codacy.com/app/mmoayyed/cas/file/2644137382/issues/source?bid=3256115&fileBranchId=3322154#l312)